htlcswitch/switch.go

```diff
@@ -2090,6 +2090,26 @@ func (s *Switch) addLiveLink(link ChannelLink) {
 	}
 	s.interfaceIndex[peerPub][link.ChanID()] = link
 
+	s.updateLinkAliases(link)
+}
+
+// UpdateLinkAliases is the externally exposed wrapper for updating link
+// aliases. It acquires the indexMtx and calls the internal method.
+func (s *Switch) UpdateLinkAliases(link ChannelLink) {
+	s.indexMtx.Lock()
+	defer s.indexMtx.Unlock()
+
+	s.updateLinkAliases(link)
+}
+
+// updateLinkAliases updates the aliases for a given link. This will cause the
+// htlcswitch to consult the alias manager on the up to date values of its
+// alias maps.
+//
+// NOTE: this MUST be called with the indexMtx held.
+func (s *Switch) updateLinkAliases(link ChannelLink) {
+	linkScid := link.ShortChanID()
+
```

Comment:
Yeah this was a bigger change we considered at the time, it partially was spun off into this issue: https://github.com/lightningnetwork/lnd/issues/8969. 

Right now the switch maintains a series of maps to mirror the state of the alias manager. Instead the alias manager should fully encapsulate all that state, ensuring that we only have a single source of truth. 

I don't think we're able to commit to making this architectural change in the scope of this PR, given the existing logic here, and the downstream items that we still need to address (cherry pick on top of 0.18.3 after merging into master, merging in changes into the two branches, etc). 

I think @GeorgeTsagk may have a side branch where he set out to bite the bullet and make the larger change.

Reply:
Ok. I'll defer regarding scope but as long as we are in agreement that this is Bad™ then I can lay this to rest. That said it does make it hard to assess the correctness of the change since we cannot reason locally about the behavior and I don't have a deep knowledge of how this state synchronizes using the disk.